@here/harp-mapview/lib/Utils.ts

```diff
@@ -146,22 +147,35 @@ export namespace MapViewUtils {
         pitchDeg: number,
         mapView: MapView
     ): geoUtils.GeoCoordinates {
-        // Get the world distance between the target and the camera.
         const pitchRad = THREE.Math.degToRad(pitchDeg);
-        const cameraHeight = distance * Math.cos(pitchRad);
-        const projectedDistanceOnTheGround = cameraHeight * Math.tan(pitchRad);
-
-        // Get the camera coordinates.
         const yawRad = THREE.Math.degToRad(yawDeg);
-        const worldTargetCoordinates = mapView.projection.projectPoint(targetCoordinates);
-        const cameraWorldCoordinates = {
-            x: worldTargetCoordinates.x + Math.sin(yawRad) * projectedDistanceOnTheGround,
-            y: worldTargetCoordinates.y - Math.cos(yawRad) * projectedDistanceOnTheGround,
-            z: 0
-        };
+        mapView.projection.projectPoint(targetCoordinates, vector);
+        const groundDistance = distance * Math.sin(pitchRad);
+        if (mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            vector.set(
+                vector.x + Math.sin(yawRad) * groundDistance,
+                vector.y - Math.cos(yawRad) * groundDistance,
+                0
+            );
+        } else if (mapView.projection.type === geoUtils.ProjectionType.Spherical) {
+            // In globe yaw and pitch are understood to be in tangent space. The approach below is
```

Comment:
Would appreciate if this was broken down into line-by-line comments for clearer comprehension.